dec2024/client.js

```diff
@@ -0,0 +1,132 @@
+require("isomorphic-fetch");
+
+class Client {
+  constructor({ hasuraSecret, targetURL }) {
+    this.secret = hasuraSecret;
+    this.url = targetURL;
+  }
+
+  async graphQL(query, variables) {
+    const result = await fetch(this.url, {
+      method: "POST",
+      body: JSON.stringify({ query, variables }),
+      headers: {
+        "x-hasura-admin-secret": this.secret,
+      },
+    });
+    return result.json();
+  }
+
+  async getQueuedFlow() {
+    const { data, errors } = await this.graphQL(
+      `query GetQueuedFlow {
+        temp_data_migrations_audit(
+          limit: 1, 
+          where: {updated: {_eq: false}}
+        ) {
+          id: flow_id
+          flow {
+            slug
+            team {
+              slug
+            }
+            data
+            publishedFlows: published_flows(
+              limit: 1, 
+              order_by: {created_at: desc}
+            ) {
+              id  
+              data
+            }
+            sessions: lowcal_sessions(
+              where: {
+                locked_at: {_is_null: true},
+                submitted_at: {_is_null: true}, 
+                sanitised_at: {_is_null: true},
+                deleted_at: {_is_null: true}
```

Comment:
Am I forgetting a simpler `where` condition for identifying "active" sessions or is this most straight-forward?

Reply:
This is correct and how I would have approached it - "active" is defined by omission as all the other conditions came later to define what "inactive" was. It's actually a very good point that this is maybe a bit more complex than it needs to be.

Reply:
Maybe a nice future feature would be a computed `is_active` column which is basically referencing these values.